packages/styled-components/src/utils/flatten.js

```diff
@@ -64,20 +62,9 @@ export default function flatten(
 
   /* Either execute or defer the function */
   if (isFunction(chunk)) {
-    if (isStatelessFunction(chunk) && executionContext) {
-      const result = chunk(executionContext);
-
-      if (process.env.NODE_ENV !== 'production' && isElement(result)) {
-        // eslint-disable-next-line no-console
-        console.warn(
-          `${getComponentName(
-            chunk
-          )} is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.`
-        );
-      }
-
-      return flatten(result, executionContext, styleSheet, stylisInstance);
-    } else return chunk;
+    return isStatelessFunction(chunk) && executionContext
+      ? flatten(chunk(executionContext), executionContext, styleSheet, stylisInstance)
```

Comment:
_Note:_ This is the only "purposeful regression" that is reasonably unexpected here, since it removes a very obvious warning from the library. I'd argue it's not really a useful one, since it'll fail in obvious ways, but since interpolations can now return objects there isn't really an alternative for this. We could reuse the check from `isPlainObject` here though if we wanted to bring it back later.

Reply:
> since it'll fail in obvious ways

I don't feel like this is an obvious fail... :confused: